public/components/Flyout/flyouts/createDetector.js

```diff
@@ -0,0 +1,833 @@
+      context.suggestedChanges.hasOwnProperty('filter_query')
+    ) {
+      return (
+        "*Detector interval recommendation hasn't been made since query filter returns no hits, you can " +
+        'either choose to continue creation without validation or try to fix filter_query'
+      );
+    } else {
+      return '';
+    }
+  };
+
+  const renderBetweenAnd = (valuess) => {
+    const values = valuess;
+    return (
+      <EuiFlexGroup alignItems="center">
+        <EuiFlexItem>
+          <FormikFieldNumber
+            name="where.fieldRangeStart"
+            fieldProps={{
+              validate: (value) => validateRange(value, values.where),
+            }}
+            inputProps={{ onChange: handleChangeWrapper, isInvalid }}
+          />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiText textAlign="center">TO</EuiText>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <FormikFieldNumber
+            name="where.fieldRangeEnd"
+            fieldProps={{
+              validate: (value) => validateRange(value, values.where),
+            }}
+            inputProps={{ onChange: handleChangeWrapper, isInvalid }}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  };
+
+  const renderValueField = (fieldType, fieldOperator) => {
+    if (fieldType == DATA_TYPES.NUMBER) {
+      return isRangeOperator(fieldOperator) ? (
+        renderBetweenAnd()
+      ) : (
+        <FormikFieldNumber
+          name="where.fieldValue"
+          fieldProps={{ validate: required }}
+          inputProps={{ onChange: handleChangeWrapper, isInvalid }}
+        />
+      );
+    } else if (fieldType == DATA_TYPES.BOOLEAN) {
+      return (
+        <FormikSelect
+          name="where.fieldValue"
+          fieldProps={{ validate: required }}
+          inputProps={{ onChange: handleChangeWrapper, options: WHERE_BOOLEAN_FILTERS, isInvalid }}
+        />
+      );
+    } else {
+      return (
+        <FormikFieldText
+          name="where.fieldValue"
+          fieldProps={{ validate: required }}
+          inputProps={{ onChange: handleChangeWrapper, isInvalid }}
+        />
+      );
+    }
+  };
+
+  async function queryMappings(index) {
+    if (!index.length) {
+      return {};
+    }
+    try {
+      const response = await context.httpClient.post('../api/alerting/_mappings', { index });
+      if (response.data.ok) {
+        return response.data.resp;
+      }
+      return {};
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  const handleOperatorChange = (e, field) => {
+    field.onChange(e);
+  };
+
+  const indexFields = getIndexFields(dataTypes, ['number', 'text', 'keyword', 'boolean']);
+  const onAddFilterButton = () =>
+    setFilterPopoverOpen((isFilterPopoverOpen) => !isFilterPopoverOpen);
+  const closePopover = () => setFilterPopoverOpen(false);
+  return {
+    flyoutProps: {
+      'aria-labelledby': 'createDetectorFlyout',
+      maxWidth: 900,
```

Comment:
Is there a reason for specifying the `maxWidth` here? Seems that the `size` prop should be sufficient here